lib/elixir/lib/calendar/iso.ex

```diff
@@ -697,16 +737,16 @@ defmodule Calendar.ISO do
 
   """
   @doc since: "1.8.0"
-  @spec day_of_era(year, month, day) :: {day :: pos_integer(), era :: 0..1}
+  @spec day_of_era(year, month, day) :: {day, era}
   @impl true
   def day_of_era(year, month, day)
-      when is_integer(year) and is_integer(month) and is_integer(day) and year > 0 do
+      when is_year_CE(year) and is_month(month) and is_day(day) do
     day = date_to_iso_days(year, month, day) - @iso_epoch + 1
     {day, 1}
   end
 
   def day_of_era(year, month, day)
-      when is_integer(year) and is_integer(month) and is_integer(day) and year < 1 do
+      when is_year_BCE(year) and is_month(month) and is_day(day) do
```

Comment:
Let's remove the guards here since we already check them on `date_to_iso_days` (if not, we should).